src/plugins/intel_npu/src/plugin/src/plugin.cpp

```diff
@@ -222,385 +204,59 @@ Plugin::Plugin()
 
     OV_ITT_TASK_CHAIN(PLUGIN, itt::domains::NPUPlugin, "Plugin::Plugin", "NPUBackends");
     _backends = std::make_shared<NPUBackends>(backendRegistry, _globalConfig);
-    OV_ITT_TASK_NEXT(PLUGIN, "registerOptions");
-    _backends->registerOptions(*_options);
+
+    /// Fetch CID version and create properties list
+    OV_ITT_TASK_NEXT(PLUGIN, "FetchCompilerVer");
+    compilerVersion cid_ver = {0, 0, 0};
```

Comment:
0 0 0 is the default state? What is happening if this is the version? I mean offline compilation in case of MLIR compiler and the device is not found.

Maybe more, what is happening if MLIR is used but CID used is an older version?

Reply:
compiler version {0,0,0} will just register all available options since all options will have version > 0. This is the case for MLIR as it will compile with the current header.